### PR TITLE
Support for Visual Studio Code & the Remote-Containers Extension

### PR DESCRIPTION
**Pull request by pbonh**
I've added support for Visual Studio Code and it's Remote-Containers extension, to make development setup a bit easier, for example:

https://dev.to/ctison/vscode-remote-containers-5740

The PR is based off of Microsoft's own container for Rust development, which uses a Debian container underneath: 
https://github.com/microsoft/vscode-remote-try-rust

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# See here for image contents: https://github.com/microsoft/vscode-dev-containers/tree/v0.134.0/containers/rust/.devcontainer/base.Dockerfile
+FROM mcr.microsoft.com/vscode/devcontainers/rust:0-1
+
+# [Optional] Uncomment this section to install additional packages.
+# RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+#     && apt-get -y install --no-install-recommends <your-package-list-here>
```
Comment (sampsyo) [inline .devcontainer/Dockerfile:6]: Delete this stuff, since we're doing this below?

**Review by sampsyo — COMMENTED** [review]
Awesome; thank you for adding this! I tried it out locally and it worked great. 🚀 

Just a tangential thought that should not block merging… I recall that @sgpthomas once upon a time made a generic Dockerfile for Calyx. Maybe we should eventually add that and then sync up this devcontainer Dockerfile to match it (in terms of dependencies)?

**Comment by rachitnigam** [comment]
Thanks for contributing @pbonh! Please ping the thread when you address @sampsyo's comment and we can merge it in.

**Comment by pbonh** [comment]
Done, thanks @sampsyo for the feedback.

**Comment by sampsyo** [comment]
Looks great; thank you again!

**Pull request merged**